src/improvements/tasks/MultisigTask.sol

```diff
@@ -793,26 +810,42 @@ abstract contract MultisigTask is Test, Script, ITask {
         vm.stopPrank();
 
         /// roll back all state changes made during the task
-        require(vm.revertTo(_startSnapshot), "failed to revert back to snapshot, unsafe state to run task");
+        require(
+            vm.revertTo(_startSnapshot), "MultisigTask: failed to revert back to snapshot, unsafe state to run task"
+        );
 
         _processStateDiffChanges(accountAccesses);
 
+        /// there should be at least one account access
+        require(accountAccesses.length > 0, "MultisigTask: no account accesses found");
+
+        /// get the minimum depth of the calls, we only care about the top level calls
+        /// this is to avoid counting subcalls as actions.
+        /// the account accesses are in order of the calls, so the first one is always the top level call
+        uint256 topLevelDepth = accountAccesses[0].depth;
+
         for (uint256 i = 0; i < accountAccesses.length; i++) {
             /// store all gnosis safe storage accesses that are writes
             for (uint256 j = 0; j < accountAccesses[i].storageAccesses.length; j++) {
-                if (accountAccesses[i].account == multisig && accountAccesses[i].storageAccesses[j].isWrite) {
+                if (accountAccesses[i].account == parentMultisig && accountAccesses[i].storageAccesses[j].isWrite) {
                     _accountAccesses.push(accountAccesses[i].storageAccesses[j]);
                 }
             }
 
-            /// only care about top level calls from the multisig,
-            /// static calls are ignored,
             /// calls to and from Addresses and the vm contract are ignored
-            /// ignore calls to vm in the build function
+            bool accountIsNotAddressesOrVm =
+                accountAccesses[i].account != address(addresses) && accountAccesses[i].account != address(vm);
+            bool accessorIsNotAddresses = accountAccesses[i].accessor != address(addresses);
+            /// only care about calls or top leveldelegate calls from the multisig, static calls are ignored
+            bool isCall = accountAccesses[i].kind == VmSafe.AccountAccessKind.Call;
+            bool isTopLevelDelegateCall = accountAccesses[i].kind == VmSafe.AccountAccessKind.DelegateCall
+                && accountAccesses[i].depth == topLevelDepth;
+            /// only record actions from the parent multisig
+            bool accessorIsParentMultisig = accountAccesses[i].accessor == parentMultisig;
+
             if (
-                accountAccesses[i].account != address(addresses) && accountAccesses[i].account != address(vm)
-                    && accountAccesses[i].accessor != address(addresses)
-                    && accountAccesses[i].kind == VmSafe.AccountAccessKind.Call && accountAccesses[i].accessor == multisig
+                accountIsNotAddressesOrVm && accessorIsNotAddresses && (isCall || isTopLevelDelegateCall)
+                    && accessorIsParentMultisig
```

Comment:
I think the `Action` struct will need to be updated to also specify whether the parent multisig is executing a call or delegatecall, and the description correspondingly updated?

Reply:
update in new PR that builds on this one at commit: https://github.com/ethereum-optimism/superchain-ops/pull/606/commits/62456eb46ff582e38d906a743fcca1e4ec3c8540